File: bank-app/app/src/main/java/com/android/bankapp/statements/StatementsPresenter.java
package com.android.bankapp.statements;

import com.android.bankapp.service.BankService;
import com.android.bankapp.service.ServiceGenerator;

import java.lang.ref.WeakReference;

class StatementsPresenter implements StatementPresenterInput {

    WeakReference<StatementsActivity> output;
    private BankService service;

    public StatementsPresenter() {
        service = ServiceGenerator.generateService(BankService.class);
    }

    @Override
    public void loadData() {

        

    }
}

File: bank_app/app/src/main/java/dev/vitorpacheco/solutis/bankapp/statementsScreen/StatementsPresenter.kt
package dev.vitorpacheco.solutis.bankapp.statementsScreen

import dev.vitorpacheco.solutis.bankapp.extensions.format
import java.lang.ref.WeakReference

interface StatementsPresenterInput {
    fun presentStatementsData(response: StatementsResponse?)
}

class StatementsPresenter : StatementsPresenterInput {

    var output: WeakReference<StatementsActivityInput>? = null

    override fun presentStatementsData(response: StatementsResponse?) {
        response?.let {
            val statements = response.statementList?.map {
                StatementViewModel(
                    title = it.title,
                    desc = it.desc,
                    date = it.date.format(),
                    value = it.value?.format()
                )
            }

            output?.get()
                ?.displayStatementsData(StatementsViewModel(statements, response.error?.message))
        }
    }

    companion object {
        var TAG = StatementsPresenter::class.java.simpleName
    }

}

